src/actions/FavoritesActions.js

```diff
@@ -35,20 +35,20 @@ export const attemptToAddFavorite = ({selectedSite, currentUser}) => {
 };
 
 
-export const attemptToRemoveFavorite = ({selectedSite, currentUser}) => {
+export const attemptToRemoveFavorite = ({favoriteSiteToRemove, currentUser}) => {
 
     return async (dispatch) => {
         const favoriteMetadata = {
-            title: selectedSite.title,
-            id: selectedSite.id,
+            title: favoriteSiteToRemove.title,
+            id: favoriteSiteToRemove.id,
         };
 
-        return firebase.firestore().doc(`users/${currentUser.email}/favorites/${selectedSite.title}`)
+        return firebase.firestore().doc(`users/${currentUser.email}/favorites/${favoriteSiteToRemove.title}`)
             .delete()
             .then(() => {
                 dispatch({
                     type: FAVORITE_REMOVED,
-                    payload: {favoriteToRemove: selectedSite}
+                    payload: {favoriteToRemove: favoriteSiteToRemove}
```

Comment:
Favorites toggle can now be called from the `SitePreview` component which has no notion of `selectedSite`, so I changed the interface for these functions to just accept a site to add/remove (not necessarily the designated selected one) and we go from there.